api/server/server.go

```diff
@@ -47,13 +46,40 @@ func New(ctx context.Context, ds models.Datastore, mq models.MessageQueue, r *ru
 		Enqueue:   enqueue,
 	}
 
-	Api.Router.Use(func(c *gin.Context) {
-		ctx, _ := common.LoggerWithFields(ctx, extractFields(c))
+	s.Router.Use(prepareMiddleware(ctx))
+
+	return s
+}
+
+func prepareMiddleware(ctx context.Context) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		ctx, log := common.LoggerWithFields(ctx, extractFields(c))
+
+		for _, param := range c.Params {
```

Comment:
Because I wanted to make sure that we run those validations and store value in context only if the target route has those params (app and route).